server/oauth_service.py

```diff
@@ -0,0 +1,52 @@
+import aiocron
+import aiohttp
+import jwt
+from jwt import InvalidTokenError
+from jwt.algorithms import RSAAlgorithm
+
+from server.config import config
+
+from .core import Service
+from .decorators import with_logger
+from .exceptions import AuthenticationError
+
+
+@with_logger
+class OAuthService(Service, name="oauth_service"):
+    """
+    Service for managing the OAuth token logins and verification.
+    """
+
+    def __init__(self):
+        self.public_keys = None
+
+    async def initialize(self) -> None:
+        await self.retrieve_public_keys()
+        # crontab: min hour day month day_of_week
+        # Run every day to update public keys.
+        self._update_cron = aiocron.crontab(
+            "0 0 * * *", func=self.retrieve_public_keys
```

Comment:
It would probably be OK to run more frequently. Maybe every 10 minutes? Or implement some logic for detecting when token signatures no longer match.

Reply:
Done